### DISCUSSION TITLE
Selectively apply draco and meshopt compression

### DISCUSSION THREAD

**Discussion by marwie** [donmccurdy/glTF-Transform#925] [Ideas]
Hi @donmccurdy 

I would like to selectively choose which meshes receive which compression since we have some projects where this would be beneficial

I was just looking a bit through the code what the `draco` and `meshopt` functions do under the hood and it seems like I can't see a way right now to filter that using the builtin methods (nor do the extension implementations for writing have a way to filter or dismiss a specific prim). 

Would you say filtering out prims would be a feature you'd be open to add/accept as a PR to gltf-transform (e.g. skip prims based on x in `draco-mesh-compression`)   
or would you prefer if users would in cases like these directly make the calls to e.g. `encodeGeometry` etc.   
or is there a third option I'm not thinking of right now?

**Comment by donmccurdy** [comment]
Hi! You're correct that the implementation doesn't provide a way to pick and choose meshes for compression at this time.

In theory the draco and meshopt encoders are written to check a list of accessors that have already been written, and then skip those accessors. That would be a basic requirement for this feature. In practice, that behavior may or may not yet work as intended, and the possibility of primitives sharing accessors is a tricky case.

I'd be open to reviewing a PR for this feature, and a decision on whether to merge or not would probably be dependent on how complex the maintenance will be. In the interest of identifying potential problems early, the API for picking primitives could just be something quick and hacky for now, like having draco skip every 2nd mesh.

**Reply by marwie** [reply]
Im trying to follow the documentation on how to setup the local CLI package and use it and seem to miss something:

- did run `npm run watch` in the root workspace directory, it's running and updating when I make codechanges
- did run `yarn link` in the cli package directory (it says it's linked but how do I verify it's not using my globally installed version?)
- when changing code (adding logs, removing something) and running `gltf-transform draco "input" "output"` it still seems to use the globally installed version. I also tried running `npm run dist`

~~What am I missing? 😅~~

Update: Working with running `node cli.js` in the dist folder for now 😊

**Reply by marwie** [reply]
Started a very minimal PR here: https://github.com/donmccurdy/glTF-Transform/pull/929 with draco skipping every second prim as you suggested and trying to do the same for meshopt (without success yet)
